src/pages/studyView/table/MultiSelectionTable.tsx

```diff
@@ -190,25 +193,35 @@ export class MultiSelectionTable extends React.Component<
                 },
                 width: columnWidth,
             },
-            [MultiSelectionTableColumnKey.MOLECULAR_PROFILE]: {
+            [MultiSelectionTableColumnKey.DATA_PROFILE]: {
                 name: columnKey,
-                headerRender: () => {
-                    return (
-                        <div
-                            style={{ marginLeft: cellMargin }}
-                            className={styles.displayFlex}
-                            data-test="profile-column-header"
-                        >
-                            {columnKey}
-                        </div>
-                    );
-                },
+                headerRender: () => <div />,
                 render: (data: MultiSelectionTableRow) => {
                     return (
                         <div className={styles.labelContent}>
-                            <EllipsisTextTooltip
-                                text={data.label}
-                            ></EllipsisTextTooltip>
+                            <EllipsisTextTooltip text={data.label} />
+                            {this.props.onClickPieChartIcon &&
+                                this.props.visibleChartIds && (
+                                    <i
+                                        className="fa fa-pie-chart"
+                                        onClick={() =>
```

Comment:
How about this?
```suggestion
                                        onClick={this.onClickAddChartIcon(data)}
```
and
```
onClickAddChartIcon = (data: MultiSelectionTableRow) => () => {
    this.props.onClickAddChartIcon!(data.uniqueKey);
}
```

Reply:
@Rajat-Sirohi I think that would work as long as `this.onClickAddChartIcon` is memoized. otherwise its the same problem.

Another way to handle this would be to wrap the render body in a component that takes the callback and the data as props, and on the inside it creates its own bound method. Then you would have to add an appropriate `key` for the row, I think